src/core/segment_reader.rs

```diff
@@ -280,6 +282,94 @@ impl SegmentReader {
         Ok(inv_idx_reader)
     }
 
+    /// Returns the list of fields that have been indexed in the segment.
+    /// The field list includes the field defined in the schema as well as the fields
+    /// that have been indexed as a part of a JSON field.
+    /// The returned field name is the full field name, including the name of the JSON field.
+    pub fn fields_metadata(&self) -> crate::Result<Vec<FieldMetadata>> {
+        let mut indexed_fields: Vec<(String, Type)> = Vec::new();
+        for (field, field_entry) in self.schema().fields() {
+            let field_name = field_entry.name().to_string();
+            let is_indexed = field_entry.is_indexed();
+
+            if is_indexed {
+                indexed_fields.push((
```

Comment:
I don't think it makes sense to add the name field name for json field.

Reply:
We need to return the full name, as this is used e.g. to select a field by a user

Reply:
I understand we want _dynamic.attributes.color in the list. I don't understand why we want _dynamic.

Reply:
removed the JSON field from the list